samples/bad_irq_check.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * Wrappers to skip touching pages and test if the kernel is disabling IRQ.
```

Comment:
It would help my understanding to have more detail about the motivation behind this test inlined here too.  The commit description still seems somewhat terse IMO.